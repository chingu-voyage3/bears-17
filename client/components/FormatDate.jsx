import React from 'react';
import PropTypes from 'prop-types';

const FormatDate = props => (
  <time dateTime={props.date}>
    {new Date(props.date).toLocaleString(props.locales, props.format)}
  </time>
);

FormatDate.defaultProps = {
  date: new Date().toISOString(),
  format: {
    year: undefined,
    month: 'short',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    second: undefined,
  },
  locales: 'en-GB', // should be navigator.language maybe?
};

FormatDate.propTypes = {
  date: PropTypes.string,
  format: PropTypes.objectOf(PropTypes.string),
  locales: PropTypes.string,
};

export default FormatDate;
