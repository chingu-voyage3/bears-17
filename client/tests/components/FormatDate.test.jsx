import React from 'react';
import renderer from 'react-test-renderer';

import FormatDate from 'Components/FormatDate';

describe('FormatDate', () => {
  it('should match the snapshot', () => {
    const date = renderer.create(<FormatDate date="2017-12-14T22:43:56.069Z" />);

    expect(date.toJSON()).toMatchSnapshot();
  });
});
