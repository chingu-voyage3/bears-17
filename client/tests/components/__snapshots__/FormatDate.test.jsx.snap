// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatDate should match the snapshot 1`] = `
<time
  dateTime="2017-12-14T22:43:56.069Z"
>
  Dec 14, 10:43 PM
</time>
`;
